connectors/sources/s3.py

```diff
@@ -126,7 +133,12 @@ async def get_bucket_objects(self, bucket):
                 bucket_obj = await s3.Bucket(bucket)
                 await asyncio.sleep(0)
 
-                async for obj_summary in bucket_obj.objects.page_size(page_size):
+                if "prefix" in kwargs:
+                    objects = bucket_obj.objects.filter(Prefix=kwargs["prefix"])
```

Comment:
Do we need to apply `page_size` here?

Reply:
We can add but that will not make a big difference in terms of performance as filter is already added here for filtering out the data.

Reply:
Just curious if this is supported?

```python
objects = bucket_obj.objects.filter(Prefix=kwargs["prefix"]).page_size(page_size)
```

Reply:
yes it is.